Makefile

```diff
@@ -2,10 +2,13 @@ test: validate
 	go test -race ./...
 
 validate:
-	golangci-lint version
-	golangci-lint config verify --config golangci.yml
-	golangci-lint run --config golangci.yml
-
+	@if command -v golangci-lint >/dev/null 2>&1; then \
+		echo "golangci-lint found, running validation..."; \
+		golangci-lint version; \
+		golangci-lint config verify --config golangci.yml || exit 1; \
+	else \
+		echo "golangci-lint not found, skipping validation."; \
```

Comment:
So no error, you added an if in the workflow for computing it only on Ubuntu

I would expect an exit code here

Reply:
yes but I want to run it on my mac when typing make, yet to fail on mac ci (it's in test target for this repo) thus exit as ok